service_client/exceptions.py

```diff
@@ -0,0 +1,57 @@
+"""
+Exception handling utilities for the Assisted Service client.
+
+This module provides custom exception classes and decorators for sanitizing
+exceptions from the Assisted Service API.
+"""
+
+from typing import Callable, TypeVar, ParamSpec, Awaitable
+from functools import wraps
+
+from assisted_service_client.rest import ApiException
+from service_client.logger import log
+
+
+class AssistedServiceAPIError(Exception):
+    """Exception for Assisted Service API errors."""
+
+
+# Type variables for the decorator
+P = ParamSpec("P")
+T = TypeVar("T")
+
+
+def sanitize_exceptions(
+    operation_name: str,
+) -> Callable[[Callable[P, Awaitable[T]]], Callable[P, Awaitable[T]]]:
+    """
+    Decorate a function to sanitize exceptions from API calls.
+
+    Args:
+        operation_name: Name of the operation for logging purposes
+
+    Returns:
+        Decorated function that catches and sanitizes exceptions
+    """
+
+    def decorator(func: Callable[P, Awaitable[T]]) -> Callable[P, Awaitable[T]]:
+        @wraps(func)
```

Comment:
You can do:
`operation_name = func.__name__`
And get rid of the `peration_name: str` in line 25